jina/helloworld/fashion/executors.py

```diff
@@ -97,3 +113,58 @@ def _norm(A):
 
 def _cosine(A_norm_ext, B_norm_ext):
     return A_norm_ext.dot(B_norm_ext).clip(min=0) / 2
+
+
+class MyEvaluator(Executor):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.eval_at = 50
+        self.num_docs = 0
+        self.total_precision = 0
+        self.total_recall = 0
+
+    @property
+    def avg_precision(self):
+        return self.total_precision / self.num_docs
+
+    @property
+    def avg_recall(self):
+        return self.total_recall / self.num_docs
+
+    def _precision(self, actual, desired):
+        if self.eval_at == 0:
+            return 0.0
+        actual_at_k = actual[: self.eval_at] if self.eval_at else actual
+        ret = len(set(actual_at_k).intersection(set(desired)))
+        sub = len(actual_at_k)
+        return ret / sub if sub != 0 else 0.0
+
+    def _recall(self, actual, desired):
+        if self.eval_at == 0:
+            return 0.0
+        actual_at_k = actual[: self.eval_at] if self.eval_at else actual
+        ret = len(set(actual_at_k).intersection(set(desired)))
+        return ret / len(desired)
+
+    @requests(on='/index')
+    def skip(self, **kwargs):
+        pass
```

Comment:
```suggestion
    @requests
    def _skip(self, **kwargs):
        pass
```

`@requests` with no `on` will catch all https://github.com/jina-ai/jina/blob/devel-2.0/.github/2.0/COOKBOOK-Executor.md#default-binding-requests-without-on